library/proc_macro/src/lib.rs

```diff
@@ -1510,15 +1510,23 @@ pub mod tracked_env {
 }
 
 /// Tracked access to additional files.
-#[unstable(feature = "track_path", issue = "99515")]
+#[unstable(feature = "proc_macro_tracked_path", issue = "99515")]
 pub mod tracked_path {
+    use std::path::{Path, PathBuf};
 
-    /// Track a file explicitly.
+    /// Track a file as if it was a dependency.
+    ///
+    /// The file is located relative to the current file where the proc-macro
+    /// is used (similarly to how modules are found). The provided path is
+    /// interpreted in a platform-specific way at compile time. So, for
```

Comment:
Is it not relative to the current working directory? So if I do `File::open("a.txt")` and `tracked_path::path("a.txt")`, can that refer to a different file?

Reply:
If I understand the docs correct, this would behave similar to using the `include!()` family in the macro expansion, which indeed uses a different directory to be relative to than `File::open()` inside the proc macro. The behavior of `include!()` is almost certainly what you want. The current working directory that rustc is invoked with when using cargo is effectively undefined.